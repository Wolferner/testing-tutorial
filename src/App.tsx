import { useEffect, useState } from 'react';
import './App.css';

function App() {
	const [data, setData] = useState<string>('');
	const [toggle, setToggle] = useState<boolean>(false);

	useEffect(() => {
		const timer = setTimeout(() => {
			setData('Zig');
		}, 100);
		return () => clearTimeout(timer);
	}, []);

	const toggleHandler = () => setToggle(prev => !prev);

	return (
		<div>
			{toggle && <div data-testid='toggle-div'>Toggle is on</div>}
			{data !== '' && <div style={{ color: 'red' }}>{data}</div>}
			<h1>Hello world!</h1>
			<input type='text' placeholder='Type something here' />
			<button onClick={toggleHandler} data-testid='toggle-btn'>
				Submit
			</button>
		</div>
	);
}

export default App;
