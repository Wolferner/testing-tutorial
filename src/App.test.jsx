import { fireEvent, render, screen } from '@testing-library/react';
import { expect } from 'vitest';
import App from './App';

describe('App component', () => {
	test('render components', () => {
		render(<App />);

		const helloWorldElem = screen.getByText(/hello world!/i);
		const btn = screen.getByRole('button');
		const input = screen.getByPlaceholderText(/Type something here/i);

		expect(helloWorldElem).toBeInTheDocument();
		expect(btn).toBeInTheDocument();
		expect(input).toBeInTheDocument().toMatchSnapshot();

		// screen.debug();
	});

	test('looking of unreal text', () => {
		render(<App />);
		const helloWorldElem = screen.queryByText(/maksim/i);
		expect(helloWorldElem).toBeNull();
	});

	test('waiting async text and style test', async () => {
		render(<App />);
		const zigElement = await screen.findByText(/zig/i);
		expect(zigElement).toBeInTheDocument();
		expect(zigElement).toHaveStyle({ color: 'rgb(255, 0, 0)' });
	});

	test('waiting async text and style test', async () => {
		render(<App />);
		const btn = screen.getByTestId('toggle-btn');

		expect(screen.queryByTestId('toggle-div')).toBeNull();
		fireEvent.click(btn);
		expect(screen.queryByTestId('toggle-div')).toBeInTheDocument();
	});
});
