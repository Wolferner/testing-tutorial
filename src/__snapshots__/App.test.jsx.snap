// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > render components 1`] = `
<input
  placeholder="Type something here"
  type="text"
/>
`;

exports[`App component > waiting async text 1`] = `
<div
  style="color: red;"
>
  Zig
</div>
`;
